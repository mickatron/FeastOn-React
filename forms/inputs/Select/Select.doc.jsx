/* eslint-disable */
import React from 'react';
import Component from './Select';
import PropTypes from 'prop-types';

import classNameType from '../../../proptypes/className';

/**
 * A Select.
 *
 * 
 * @composedWith UiWrapper
*/
function Select(props) {
  return <Component {...props} />;
}

Select.propTypes = {
  /** Options to be rendered by the Select component. */
  options: PropTypes.arrayOf(PropTypes.shape({
    className: classNameType,
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
  /** The selected option value of the Select component. */
  value: PropTypes.oneOf([
    PropTypes.string,
    PropTypes.number,
    PropTypes.array,
  ]),
}

export default Select;
