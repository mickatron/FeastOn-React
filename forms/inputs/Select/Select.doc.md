```js
<Select 
  onChange={() => {}}
  options={[
    {
      label: 'Select Value 1',
      value: 'Select Value 1',
    },
    {
      label: 'Select Value 2',
      value: 'Select Value 2',
    },
    {
      label: 'Select Value 3',
      value: 'Select Value 3',
    },
    {
      label: 'Select Value 4',
      value: 'Select Value 4',
    },
  ]}
/>
```