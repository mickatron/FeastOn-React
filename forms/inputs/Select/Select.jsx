import PropTypes from 'prop-types';
import React from 'react';

import classNameType from '../../../proptypes/className';
import uiWrapper from '../../../HOC/uiWrapper/uiWrapper';

class Select extends React.PureComponent {
  static propTypes = {
    /** Options to be rendered by the Select component. */
    options: PropTypes.arrayOf(PropTypes.shape({
      className: classNameType,
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })).isRequired,
    /** The selected option value of the Select component. */
    value: PropTypes.oneOf([
      PropTypes.string,
      PropTypes.number,
      PropTypes.array,
    ]),
  }

  static defaultProps = {
    value: undefined,
  }

  render() {
    const {
      options,
      value,
    } = this.props;

    return (
      <React.Fragment>
        {options.map(option => (
          <option
            key={option.value}
            className={option.className}
            value={option.value}
            selected={value && option.value === value}
          >
            {option.label}
          </option>
        ))}
      </React.Fragment>
    );
  }
}

// return HOC wrapped component
export default uiWrapper({
  name: 'Select',
  Component: Select,
  element: 'select',
});
