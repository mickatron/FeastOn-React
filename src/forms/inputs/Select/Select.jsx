import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/shared/proptypes/className';

export default class Select extends React.PureComponent {

  static propTypes = {
    className: classNameType,
    options: PropTypes.arrayOf({
      className: classNameType,
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }).isRequired,
    value: PropTypes.oneOf([ 
      PropTypes.string,
      PropTypes.number,
      PropTypes.array,
    ]),
  }

  render() {
    const { className, options, value, ...other } = this.props;
    return (
      <select 
        className={classnames(className, 'c-Select')} 
        {...other} 
      >
        { options.map((options) => {
          const {className, label, value} = options;
          return <option className={className} value={value}>{label}</option>;
        }) }
      </select>
    );
  }
}