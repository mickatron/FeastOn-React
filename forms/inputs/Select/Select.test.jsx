import { render } from 'enzyme';
import React from 'react';
import Select from './Select';

describe('Select', () => {
  let component = null;

  it('renders', () => {
    component = render((
      <Select
        onChange={() => { }}
        options={[
          {
            label: 'Select Value 1',
            value: 'Select Value 1',
          },
          {
            label: 'Select Value 2',
            value: 'Select Value 2',
          },
          {
            label: 'Select Value 3',
            value: 'Select Value 3',
          },
          {
            label: 'Select Value 4',
            value: 'Select Value 4',
          },
        ]}
      />
    ));
    expect(component).toMatchSnapshot();
  });
});
