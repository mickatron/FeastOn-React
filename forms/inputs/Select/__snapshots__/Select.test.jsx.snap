// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "Select",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "value": "Select Value 1",
        },
        "children": Array [
          Object {
            "data": "Select Value 1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "option",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "value": "Select Value 2",
          },
          "children": Array [
            Object {
              "data": "Select Value 2",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "value": "Select Value 3",
            },
            "children": Array [
              Object {
                "data": "Select Value 3",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "option",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "value": "Select Value 4",
              },
              "children": Array [
                Object {
                  "data": "Select Value 4",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "option",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "value": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "value": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "value": undefined,
        },
      },
      Object {
        "attribs": Object {
          "value": "Select Value 2",
        },
        "children": Array [
          Object {
            "data": "Select Value 2",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "option",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "value": "Select Value 3",
          },
          "children": Array [
            Object {
              "data": "Select Value 3",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "value": "Select Value 4",
            },
            "children": Array [
              Object {
                "data": "Select Value 4",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "option",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "value": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "value": "Select Value 1",
          },
          "children": Array [
            Object {
              "data": "Select Value 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "value": undefined,
        },
      },
      Object {
        "attribs": Object {
          "value": "Select Value 3",
        },
        "children": Array [
          Object {
            "data": "Select Value 3",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "option",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "value": "Select Value 4",
          },
          "children": Array [
            Object {
              "data": "Select Value 4",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "value": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "value": "Select Value 2",
          },
          "children": Array [
            Object {
              "data": "Select Value 2",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "value": "Select Value 1",
            },
            "children": Array [
              Object {
                "data": "Select Value 1",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "option",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "value": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "value": undefined,
        },
      },
      Object {
        "attribs": Object {
          "value": "Select Value 4",
        },
        "children": Array [
          Object {
            "data": "Select Value 4",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "option",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "value": "Select Value 3",
          },
          "children": Array [
            Object {
              "data": "Select Value 3",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "option",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "value": "Select Value 2",
            },
            "children": Array [
              Object {
                "data": "Select Value 2",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "option",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "value": "Select Value 1",
              },
              "children": Array [
                Object {
                  "data": "Select Value 1",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "option",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "value": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "value": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "value": undefined,
        },
      },
    ],
    "name": "select",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
